hibernate-core/src/main/java/org/hibernate/engine/spi/CascadingAction.java

```diff
@@ -92,4 +97,16 @@ default void noCascade(EventSource session, Object parent, EntityPersister persi
 	 * Should this action be performed (or noCascade consulted) in the case of lazy properties.
 	 */
 	boolean performOnLazyProperty();
+
+	/**
+	 * The cascade direction in which we care whether the foreign key is declared with
+	 * {@link org.hibernate.annotations.OnDeleteAction#CASCADE on delete cascade}.
+	 *
+	 * @apiNote This allows us to reuse the long-existing boolean parameter of
+	 *          {@link #cascade(EventSource,Object,String,String,String, List,Object,boolean)}
+	 *          for multiple purposes.
+	 *
+	 */
+	@Incubating @Nullable
+	ForeignKeyDirection directionAffectedByCascadeDelete();
```

Comment:
I see this class is SPI. I guess it's only intended to be _called_ by users/integrators, not implemented?

Otherwise we would need to look into adding a default implementation.

Reply:
Unnecessary considering what Christian said, but if you already did it, yes it seems fine :) Thank you